src/nvim/api/vim.c

```diff
@@ -2450,3 +2451,203 @@ void nvim_del_user_command(String name, Error *err)
 {
   nvim_buf_del_user_command(-1, name, err);
 }
+
+/// Parse command line.
+///
+/// Doesn't check the validity of command arguments.
+///
+/// @param str       Command line string to parse. Cannot contain "\n".
+/// @param opts      Optional parameters. Reserved for future use.
+/// @param[out] err  Error details, if any.
+/// @return Dictionary containing command information, with these keys:
+///         - cmd: (string) Command name.
+///         - line1: (number) Starting line of command range. Only applicable if command can take a
+///                  range.
+///         - line2: (number) Final line of command range. Only applicable if command can take a
+///                  range.
+///         - bang: (boolean) Whether command contains a bang (!) modifier.
+///         - args: (array) Command arguments.
+///         - addr: (string) Value of |:command-addr|. Uses short name.
+///         - nargs: (string) Value of |:command-nargs|.
+///         - nextcmd: (string) Next command if there are multiple commands separated by a |:bar|.
+///                             Empty if there isn't a next command.
+///         - magic: (dictionary) Which characters have special meaning in the command arguments.
+///             - file: (boolean) The command expands filenames. Which means characters such as "%",
+///                               "#" and wildcards are expanded.
```

Comment:
👍  If we find exceptions where wildcards are literal but `%`/`#` are magic, we can add variants later.

Reply:
Yeah